pkg/controller/promotionreconciler/reconciler.go

```diff
@@ -32,47 +32,36 @@ import (
 )
 
 type Options struct {
-	DryRun                bool
 	CIOperatorConfigAgent agents.ConfigAgent
-	ConfigGetter          config.Getter
 	GitHubClient          github.Client
-	// The registryManager is set up to talk to the cluster
-	// that contains our imageRegistry. This cluster is
-	// most likely not the one the normal manager talks to.
-	RegistryManager controllerruntime.Manager
 }
 
 const ControllerName = "promotionreconciler"
 
-func AddToManager(mgr controllerruntime.Manager, opts Options) error {
+func AddToManager(mgr controllerruntime.Manager, opts Options, enqueuer prowjobreconciler.Enqueuer) error {
 	// Pre-Allocate the Image informer rather than letting it allocate on demand, because
 	// starting the watch takes very long (~2 minutes) and having that delay added to our
 	// first (# worker) reconciles skews the workqueue duration metric bigtimes.
-	if _, err := opts.RegistryManager.GetCache().GetInformer(context.TODO(), &imagev1.Image{}); err != nil {
+	if _, err := mgr.GetCache().GetInformer(context.TODO(), &imagev1.Image{}); err != nil {
 		return fmt.Errorf("failed to get informer for image: %w", err)
 	}
 
 	if err := opts.CIOperatorConfigAgent.AddIndex(configIndexName, configIndexFn); err != nil {
 		return fmt.Errorf("failed to add indexer to config-agent: %w", err)
 	}
 
-	prowJobEnqueuer, err := prowjobreconciler.AddToManager(mgr, opts.ConfigGetter, opts.DryRun)
-	if err != nil {
-		return fmt.Errorf("failed to construct prowjobreconciler: %w", err)
-	}
-
 	log := logrus.WithField("controller", ControllerName)
 	r := &reconciler{
 		ctx:    context.Background(),
 		log:    log,
-		client: imagestreamtagwrapper.MustNew(opts.RegistryManager.GetClient(), opts.RegistryManager.GetCache()),
+		client: imagestreamtagwrapper.MustNew(mgr.GetClient(), mgr.GetCache()),
```

Comment:
Same here, wrong `manager`

Reply:
No, I changed the one that we pass in, since we no longer needed to have both around

Reply:
Yeah but this has a set of issues:
* We are not explicit anymore about the fact that this must be the registry cluster, making it likely that we screw this up when we change the registry
* It messes with the start order, now we start main manager -> registry manager -> controller, as opposed to main manager -> controller